### ISSUE TITLE
CA2000 false positive with value tuple deconstruct

### ISSUE THREAD

**Issue by gjaw** [dotnet/roslyn-analyzers#3942]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)

#### Diagnostic ID

Example: [CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000)

#### Repro steps

Easy repro with the following code:

``` csharp
using System;
using System.IO;

namespace ca2000bug
{
    public static class Class1
    {
        public static IDisposable Example(int x)
        {
            IDisposable y;
            if (x == 0)
            {
                y = new MemoryStream(); // CA2000 false positive on this line
            }
            else
            {
                (y, _) = GetByZ(x); // This construct is the key to trigger the false positive above
            }
            return y;
        }

        private static (IDisposable, int) GetByZ(int z)
        {
            return (new MemoryStream(), z + 1);
        }
    }
}
```

#### Expected behavior

I expect no CA2000 on the specified line since the disposable object will be returned from the method, so it will not be out of scope.

#### Actual behavior

CA2000 is triggered. This only happens if the alternative code path includes assignment to the variable `y` as a value tuple deconstruct as shown in the example.
